pandas/core/frame.py

```diff
@@ -5145,36 +5145,151 @@ def pivot_table(self, values=None, index=None, columns=None,
+        ...                                     index=['one', 'two'],
+        ...                                     columns=multicol2)
+
+        Stacking a dataframe with a single level column axis returns a Series:
+
+        >>> df_single_level_cols
              a   b
-        one  1.  2.
-        two  3.  4.
+        one  0   1
+        two  2   3
+        >>> df_single_level_cols.stack()
+        one  a    0
+             b    1
+        two  a    2
+             b    3
+        dtype: int64
+
+        Stacking a dataframe with a multi-level column axis:
+
+        >>> df_multi_level_cols1
+             X
+             a   b
+        one  0   1
+        two  2   3
+        >>> df_multi_level_cols1.stack()
+                  X
+        one  a    0
+             b    1
+        two  a    2
+             b    3
+
+        It is common to have missing values when stacking a dataframe
+        with multi-level columns, as the stacked dataframe typically
+        has more values than the original dataframe. Missing values
+        are filled with NaNs:
+
+        >>> df_multi_level_cols2
+               X     Y
+               a     b
+        one  0.0   1.0
+        two  2.0   3.0
+        >>> df_multi_level_cols2.stack()
+                 X    Y
+        one a  0.0  NaN
+            b  NaN  1.0
+        two a  2.0  NaN
+            b  NaN  3.0
+
+        The first parameter controls which level or levels are stacked:
+
+        >>> df_multi_level_cols2.stack(0)
+                 a    b
+        one X  0.0  NaN
+            Y  NaN  1.0
+        two X  2.0  NaN
+            Y  NaN  3.0
+        >>> df_multi_level_cols2.stack([0, 1])
+        one  X  a    0.0
+             Y  b    1.0
+        two  X  a    2.0
+             Y  b    3.0
+        dtype: float64
 
-        >>> s.stack()
-        one a    1
-            b    2
-        two a    3
-            b    4
+        Note that rows where all values are missing are dropped by
+        default but this behaviour can be controlled via the dropna
+        keyword parameter:
+
+        >>> df_multi_level_cols3
+               X     Y
+               a     b
+        one  NaN   1.0
+        two  2.0   3.0
+        >>> df_multi_level_cols3.stack(dropna=False)
+                 X    Y
+        one a  NaN  NaN
+            b  NaN  1.0
+        two a  2.0  NaN
+            b  NaN  3.0
+
+        >>> df_multi_level_cols3.stack(dropna=True)
+                 X    Y
+        one b  NaN  1.0
+        two a  2.0  NaN
+            b  NaN  3.0
 
         Returns
         -------
-        stacked : DataFrame or Series
+        DataFrame or Series
+            Stacked dataframe or series.
+
+        See Also
+        --------
+        DataFrame.unstack: unstack prescribed level(s) from index axis
+             onto column axis.
+        DataFrame.pivot: reshape dataframe from long format to wide
+             format.
+        DataFrame.pivot_table: create a spreadsheet-style pivot table
```

Comment:
It should be a space between the colon, and the description should start with a capital letter.

Reply:
Fixed in 652f7b2